injection/sharedmain/main.go

```diff
@@ -245,3 +287,89 @@ func flush(logger *zap.SugaredLogger) {
 	logger.Sync()
 	metrics.FlushExporter()
 }
+
+// ParseAndGetConfigOrDie parses the rest config flags and creates a client or
+// dies by calling log.Fatalf.
+func ParseAndGetConfigOrDie() *rest.Config {
+	var (
+		masterURL = flag.String("master", "",
+			"The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+		kubeconfig = flag.String("kubeconfig", "",
+			"Path to a kubeconfig. Only required if out-of-cluster.")
+	)
+	flag.Parse()
+
+	cfg, err := GetConfig(*masterURL, *kubeconfig)
+	if err != nil {
+		log.Fatalf("Error building kubeconfig: %v", err)
+	}
+
+	return cfg
+}
+
+// MemStatsOrDie sets up reporting on Go memory usage every 30 seconds or dies
+// by calling log.Fatalf.
+func MemStatsOrDie(ctx context.Context) {
+	msp := metrics.NewMemStatsAll()
+	msp.Start(ctx, 30*time.Second)
+
+	if err := view.Register(msp.DefaultViews()...); err != nil {
+		log.Fatalf("Error exporting go memstats view: %v", err)
+	}
+}
+
+// SetupLoggerOrDie sets up the logger using the config from the given context
+// and returns a logger and atomic level, or dies by calling log.Fatalf.
+func SetupLoggerOrDie(ctx context.Context, component string) (*zap.SugaredLogger, zap.AtomicLevel) {
+	loggingConfig, err := GetLoggingConfig(ctx)
+	if err != nil {
+		log.Fatalf("Error reading/parsing logging configuration: %v", err)
+	}
+	logger, atomicLevel := logging.NewLoggerFromConfig(loggingConfig, component)
+	ctx = logging.WithLogger(ctx, logger)
```

Comment:
You might want to document this in the function comments. I think the intent is that this will also modify the incoming context and attach a logger to it. I think it would be good to document that because it's not (to me) immediately clear that's what's happening. Alternatively, if the intent is for the caller to then create / decorate their context with the logger is to return above with:
return logging.NewLoggerFromConfig(loggingConfig, component).

Reply:
I think the code as it stands is even wrong. `WithLogger` will create a new context that you have no access to in this case.

I agree, we should probably just do the `WithLogger` call right after calling `SetupLoggerOrDie` at the calling side.

Reply:
Whoops, good catch. I will fix